packages/stream-metadata/src/evmRpcClient.ts

```diff
@@ -1,44 +1,51 @@
+import { RiverRegistry } from '@river-build/web3'
 import { config } from './environment'
-import { createPublicClient, http } from 'viem'
-import { riverChainDevnet, riverChainLocalhost, riverChainProduction } from './evmChainConfig'
-
-const publicClient: Record<number, ReturnType<typeof createPublicClientFromChainId>> = {}
-
-function createPublicClientFromChainId(chainId: number) {
-	let riverChainUrl: string
-
-	switch (chainId) {
-		case 550:
-			riverChainUrl = riverChainProduction.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainProduction,
-				transport: http(riverChainUrl),
-			})
-		case 6524490:
-			riverChainUrl = riverChainDevnet.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainDevnet,
-				transport: http(riverChainUrl),
-			})
-		case 31338:
-			riverChainUrl = riverChainLocalhost.rpcUrls.default.http[0]
-			return createPublicClient({
-				chain: riverChainLocalhost,
-				transport: http(riverChainUrl),
-			})
-		default:
-			console.error(`Unsupported chain ID: ${chainId}`)
-			throw new Error(`Unsupported chain ${chainId}`)
+import { ethers } from 'ethers'
+
+const chainConfigs: ChainConfigs = {
+	550: {
+		rpcUrl: 'https://mainnet.rpc.river.build/http',
+	},
+	6524490: {
+		rpcUrl: 'https://devnet.rpc.river.build',
+	},
+	31338: {
+		rpcUrl: 'http://127.0.0.1:8546',
+	},
+}
+
+interface ChainConfigs {
+	[chainId: number]: ChainConfig
+}
+
+type ChainConfig = {
+	rpcUrl: string
+}
+
+const riverRegistry: Record<number, ReturnType<typeof createRiverRegistry>> = {}
+
+function createRiverRegistry(chainId: number) {
+	const riverChainUrl = chainConfigs[chainId]?.rpcUrl
+	if (!riverChainUrl) {
+		throw new Error(`Unsupported chain ${chainId}`)
+	}
+
+	const provider = new ethers.providers.JsonRpcProvider(riverChainUrl)
+	const riverRegistry = new RiverRegistry(config.river, provider)
+
+	if (!riverRegistry) {
+		throw new Error('Failed to create river registry')
 	}
+
+	return riverRegistry
 }
 
-export function getPublicClient() {
-	const chainId = config.chainId
+export function getRiverRegistry(chainId: number) {
```

Comment:
this is weird - we don't have 1 river registry per chain, we have one river config per environment, which includes a chain id and contract addresses. You should just look at `config.river.chainId`